File: JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands/lobby/career/CheckDisplayExistCommand.java
package bluffinmuffin.protocol.commands.lobby.career;

import java.util.StringTokenizer;

import bluffinmuffin.protocol.commands.Command;
import bluffinmuffin.protocol.commands.ICommand;

public class CheckDisplayExistCommand implements ICommand
{
    public static String COMMAND_NAME = "lobbyCAREER_CHECK_DISPLAY_EXIST";
    
    private final String m_displayName;
    
    public CheckDisplayExistCommand(StringTokenizer argsToken)
    {
        m_displayName = argsToken.nextToken();
    }
    
    public CheckDisplayExistCommand(String p_DisplayName)
    {
        m_displayName = p_DisplayName;
    }
    
    @Override
    public String encodeCommand()
    {
        final StringBuilder sb = new StringBuilder();
        sb.append(CheckDisplayExistCommand.COMMAND_NAME);
        sb.append(Command.L_DELIMITER);
        sb.append(m_displayName);
        sb.append(Command.L_DELIMITER);
        return sb.toString();
    }
    
    public String encodeResponse(Boolean exist)
    {
        return new CheckDisplayExistResponse(m_displayName, exist).encodeCommand();
    }
    
    public String getDisplayName()
    {
        return m_displayName;
    }
}

File: C#/Com.Ericmas001.Game.Poker.Protocol.Commands/Lobby/Career/CheckDisplayExistCommand.cs
using System;
using System.Collections.Generic;
using System.Text;
using Com.Ericmas001;
using Com.Ericmas001.Net.Protocol.JSON;
using Newtonsoft.Json.Linq;

namespace Com.Ericmas001.Game.Poker.Protocol.Commands.Lobby.Career
{
    public class CheckDisplayExistCommand : AbstractLobbyCommand
    {
        public static string COMMAND_NAME = "lobbyCAREER_CHECK_DISPLAY_EXIST";

        public string DisplayName { get; set; }

        public string EncodeResponse(bool exist)
        {
            return new CheckDisplayExistResponse(this) { Exist = exist }.Encode();
        }
    }
}
